pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
+	serverVersionNumber := seg[0][1:]
+	serverCommit := binaryInfo.GitCommit
+	if len(serverCommit) > 7 {
+		serverCommit = serverCommit[:7]
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method != "PUT" && req.Method != "POST" {
```

Comment:
> No, skewed usage checks are not only useful for stripping logic, they are
> primarily used by admins to force clients to upgrade because the logic the
> client is using in the patch is faulty.  The *goal* of skew checks is to
> deny clients.  *sometimes* the reason a client is denied is because it
> doesn't follow correct logic.

This is an outdated diff because you can now choose which verbs you want to apply the check to, so its possible to operate against PATCH.

I still maintain that PATCH is safe and that only really, really, really borked clients will generate patches that valid, but have unintended behavior.